### ISSUE TITLE
Underscore as an attribute is not rendered - used with htmx/hyperscript

### ISSUE THREAD

**Issue by pablo8itall** [Knio/dominate#193]
When trying to generate html that uses [hyperscript](https://hyperscript.org/) in my htmx webapp Dominate does not allow me to use _ as an attribute for a tag.

```
>>>tag = div(_='on click do something')
>>>tag.render()  
'<div ="on click do something"></div>'
```
I assume its something to do with Python as typically we'd ignore named variables when they are underscores.

```
>>> def test(_):
...     print(_)
... 
>>> test(1)
1
>>> 
```

Although this just seems to be a convention and not a hard coded language feature.

**Comment by pablo8itall** [comment]
Works if you do this:
```
>>> tag['_']='on fix do fixin'
>>> tag.render()
'<div ="on click do something" _="on fix do fixin"></div>'
```

**Comment by Knio** [comment]
This is actually from [here](https://github.com/Knio/dominate/blob/master/dominate/dom_tag.py#L444-L445), as a way to allow leading underscores to attribute names that would otherwise collide with Python reserved words.

**Comment by pablo8itall** [comment]
Regarding how Dominate handles the hyphen and underscore generally. If I add an attribute with an underscore, when is it converted to a hyphen. I see that data_something will convert to data-something at render.

This is important for htmx as it uses hx- for all its attributes.

**Comment by Knio** [comment]
yes, `hx_` could be added there. Or that logic could be revisited (how many html attrs use underscores?)

**Comment by angelsen** [comment]
Adding `hx_` would handle htmx tags as they are `hx-sometag`.

```python
# Workaround for dash
special_prefix = any([attribute.startswith(x) for x in ('data_', 'aria_', 'hx_')])
if attribute in set(['http_equiv']) or special_prefix:
  attribute = attribute.replace('_', '-').lower()
  ```
  
  handle:
  
  ```python
  bst.form_control(
    placeholder="Begin typing to search",
    name="search",
    hx_post=url("search", 1),
    hx_trigger="load, input changed delay:500ms, search",
    hx_target=f"#{id}",
    hx_headers=csrf_header(request),
),
```

**Comment by hasansezertasan** [comment]
> I'm switching to Dominate from Jinja templates, and its a good deal faster and cleaner for me.

How is that going, if it's an open source project, please live a link.

**Comment by nikalexis** [comment]
Have a look also in this pull request for HTMX (and some more) helpers / directives.

https://github.com/Knio/dominate/pull/203

**Comment by Knio** [comment]
Fixed in #210
